app/models/dialog_field_text_box.rb

```diff
@@ -38,10 +38,12 @@ def validate_field_data(dialog_tab, dialog_group)
 
     return "#{dialog_tab.label}/#{dialog_group.label}/#{label} is required" if required? && value.blank?
 
-    case validator_type
-    when 'regex'
-      return "#{dialog_tab.label}/#{dialog_group.label}/#{label} is invalid" unless value.match(/#{validator_rule}/)
-    end
+    # currently only regex is supported
+    rule = validator_rule if validator_type == 'regex'
+    rule ||= "^[0-9]+$" if data_type == "integer"
```

Comment:
@gmcculloug @bzwei Prefer `\A` and `\z` over `^` and `$` for validations.  This is because `^` and `$` only look at a single line and ignore \n boundaries.

Compare:

```ruby
[1] pry(main)> !!"`rm -rf /`\n1234".match(/^[0-9]+$/)
=> true
[1] pry(main)> !!"`rm -rf /`\n1234".match(/\A[0-9]+\z/)
=> false
```